pkg/obfuscate/sql_tokenizer.go

```diff
@@ -387,6 +387,9 @@ func (tkn *SQLTokenizer) Scan() (TokenKind, []byte) {
 			if kind == DollarQuotedFunc {
 				// this is considered an embedded query, we should try and
 				// obfuscate it
+				// Trim off the $func$ tag
```

Comment:
Why? The `$func$` tag is just like a quote, which you chose to keep in this PR.

Reply:
The contents of a `$func$`-enclosed string is SQL. The tag needs to be trimmed off here so that the contents (without the `$func$` tag) can be obfuscated. It gets added back on at the end of this block.

Reply:
I'm not sure I understand. Sounds to me like there was a bug previously? The query didn't get obfuscated? Regardless, this seem unrelated to the scope of your PR, so maybe we should treat it separately.

Reply:
My understanding is that function code is declared within dollar-quoted strings, but other strings can also be: any Postgres string can use this quote style

Reply:
Correct. Any string can be in there, and there are various feature flags on handling these.

Reply:
This is just preserving the existing behavior. Previously the functions returned the strings without the quote marks (including without the `$func$`). Now, they include the quote marks. Since they didn't have them before, they didn't need to be trimmed in order to parse the SQL function. Now the quotes are included and so need to be trimmed and replaced when parsing the function.

Reply:
I'm not sure I understand this either. Maybe it's something that can be described in the commit message?

Reply:
> This is just preserving the existing behavior. Previously the functions returned the strings without the quote marks (including without the $func$). Now, they include the quote marks. Since they didn't have them before, they didn't need to be trimmed in order to parse the SQL function. Now the quotes are included and so need to be trimmed and replaced when parsing the function.

Ok that makes sense, but lease apply Katie's [suggestion](https://github.com/DataDog/datadog-agent/pull/11967#discussion_r873630767). Perhaps the `$func` string should be a constant (something like `funcDelimiter` or `delimiterFunc`).